omero/sysadmins/omeroweb-upgrade.rst

```diff
@@ -0,0 +1,155 @@
+
+Plugin updates
+^^^^^^^^^^^^^^
+
+OMERO.web plugins are very closely integrated into the webclient. For this
+reason, it is possible that an update of OMERO will cause issues with an older
+version of a plugin. It is best when updating the server to also install any
+available plugin updates according to their own documentation.
+
+Since 5.3, all official OMERO.web plugins can be installed from :pypi:`Python Package Index <>`.
+You should remove all previously installed plugins and install the latest
+versions using `pip <https://pip.pypa.io/en/stable/>`_.
+
+Also introduced in 5.3, the ``Open with`` configuration allows users to open data in other web applications
+e.g. open images in a custom viewer or open images in a new figure with OMERO.figure.
+After installing OMERO.figure (or any other app), run the following command to add it to the ``Open with`` options, 
+so that the app is available from the context menu on the webclient tree::
+
+    $ bin/omero config append omero.web.open_with '["omero_figure", "new_figure",
+      {"supported_objects":["images"], "target": "_blank", "label": "OMERO.figure"}]'
+
+Updating 'Open with' config
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you have configured :property:`omero.web.open_with` prior to OMERO 5.3.3 and
+also set the default viewer with :property:`omero.web.viewer.view`, for example
+as described for `OMERO.iviewer <https://pypi.python.org/pypi/omero-iviewer>`_
+then you will find that ``Open with > Image Viewer`` also opens the OMERO.iviewer
+instead of the ``webgateway`` viewer.
+
+To fix this, you need to update the ``Image Viewer`` option within
+your :property:`omero.web.open_with` config.
+
+The best way to do this without changing the ordering of the options is to
+``get`` the complete current config, edit the ``Image Viewer`` option, replacing
+``"webindex"`` with ``"webgateway"`` and then ``set`` this as the updated config::
+
+    $ bin/omero config get omero.web.open_with
+    [["Image viewer", "webindex", {"supported_objects": ["image"], "script_url": "we....
+
+    # Replace "webindex" with "webgateway" and paste everything back to set, within single quotes
+
+    $ bin/omero config set omero.web.open_with '[["Image viewer", "webgateway", {"supported_objects": ["image"], "scr....'
+
+Troubleshooting
+^^^^^^^^^^^^^^^
+
+If you encounter errors during an OMERO.web upgrade, etc., you
+should retain as much log information as possible and notify the OMERO.server
+team via the mailing lists available on the :community:`support <>`
+page.
+
+
+Update your OMERO.web server configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+FastCGI support was removed in OMERO 5.2 and OMERO.web can be deployed
+using WSGI (see :ref:`omero_web_deployment` for more details).
+If you have already deployed OMERO.web using WSGI you should regenerate your
+config files, remembering to merge in any of your own modifications if
+necessary. **Due to the nature of OMERO.web development for the 5.2.x line,
```

Comment:
5.2? Do we care about that on this branch?

Reply:
It can be removed now

Reply:
Second though, some people might have skipped 5.3 and do the jump to 5.4 (since they will be very closed) so safer to keep it for this round

Reply:
Maybe reword to make it clearer that it actually still applies to 5.4?

Reply:
maybe "due to changes made in ..." or suchlike